src/test/java/cloud/fogbow/ras/core/plugins/interoperability/aws/publicip/v2/AwsV2PublicIpPluginTest.java

```diff
@@ -0,0 +1,678 @@
+				.thenThrow(SdkClientException.builder().build());
+
+		String allocationId = FAKE_ALLOCATION_ID;
+
+		// exercise
+		this.plugin.doCreateSecurityGroups(allocationId, client);
+	}
+	
+	// test case: When calling the doAuthorizeSecurityGroupIngress method, and an
+	// error occurs during the request, an UnexpectedException will be thrown.
+	@Test(expected = UnexpectedException.class) // verify
+	public void testDoAuthorizeSecurityGroupIngressUnsuccessful() throws FogbowException {
+		// set up
+		Ec2Client client = Mockito.mock(Ec2Client.class);
+		PowerMockito.mockStatic(AwsV2ClientUtil.class);
+		BDDMockito.given(AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+
+		Mockito.when(client.authorizeSecurityGroupIngress(Mockito.any(AuthorizeSecurityGroupIngressRequest.class)))
+				.thenThrow(SdkClientException.builder().build());
+
+		String allocationId = FAKE_ALLOCATION_ID;
+		String groupId = FAKE_DEFAULT_SECURITY_GROUP_ID;
+
+		// exercise
+		this.plugin.doAuthorizeSecurityGroupIngress(allocationId, groupId, client);
+	}
+	
+	// test case: When calling the doDeleteSecurityGroups method, and an error
+	// occurs during the request, an UnexpectedException will be thrown.
+	@Test(expected = UnexpectedException.class) // verify
+	public void testDoDeleteSecurityGroupsUnsuccessful() throws FogbowException {
+		// set up
+		Ec2Client client = Mockito.mock(Ec2Client.class);
+		PowerMockito.mockStatic(AwsV2ClientUtil.class);
+		BDDMockito.given(AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+
+		Mockito.when(client.deleteSecurityGroup(Mockito.any(DeleteSecurityGroupRequest.class)))
+				.thenThrow(SdkClientException.builder().build());
+
+		String groupId = FAKE_DEFAULT_SECURITY_GROUP_ID;
+
+		// exercise
+		this.plugin.doDeleteSecurityGroups(groupId, client);
+	}
+	
+	// test case: When calling the doCreateTagsRequests method, without a valid
+	// tag request, an UnexpectedException will be thrown.
+	@Test(expected = UnexpectedException.class) // verify
+	public void testDoCreateTagsRequestsUnsuccessful() throws FogbowException {
+		// set up
+		Ec2Client client = Mockito.mock(Ec2Client.class);
+		PowerMockito.mockStatic(AwsV2ClientUtil.class);
+		BDDMockito.given(AwsV2ClientUtil.createEc2Client(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+
+		Mockito.when(client.createTags(Mockito.any(CreateTagsRequest.class)))
+				.thenThrow(SdkClientException.builder().build());
+
+		String key = null;
+		String value = null;
+		String resourceId = null;
+
+		// exercise
+		this.plugin.doCreateTagsRequests(key, value, resourceId, client);
+	}
+	
+	// test case: When calling the selectNetworkInterfaceFrom method, without anyone
+	// sub-net ID different from the default sub-net, it must return the network
+	// interface of default sub-net.
+	@Test
+	public void testSelectNetworkInterfaceUnsuccessful() throws FogbowException {
```

Comment:
shouldn't it return null? how the others plugins work?

Reply:
In other plug-ins, such as OpenStack and CloudStack, public IP is attached using to the default network, which may be inaccurate compared to what we discussed earlier.